src/GeositeFramework/js/widgets/map_utils/measure/AgsMeasure.js

```diff
@@ -261,10 +284,21 @@ define([
 
                 // Calculate the length of the line, using a geographic coordindate system
                 geographicLine = webMercatorUtils.webMercatorToGeographic(line);
-                var geoLineLength = geodesicUtils.geodesicLengths([geographicLine], options.esriLengthUnits)[0];
+                var geoLineLength = getLength(geographicLine);
 
                 // Format the segment and line lengths
-                tipText = formatTooltip(geoLineLength, _renderedLength + geoLineLength);
+                var totalLength = _.map(
```

Comment:
I find this block pretty terse and hard to follow.  Breaking it down to component parts is about equivilent length but much better to read, in my opinion.  Along the lines of
```js
                var units = _.map(options.esriLengthUnits, function(unit) {
                        return _unitsLookup[unit];
                    }),
                    allSegments = _renderedLength.concat(geoLineLength),
                    totalLength = _.map(units, function(unit) {
                        return {
                            uom: unit,
                            length: sum(_.where(allSegments, { uom: unit }))
                        };
                    });
                function sum(arr) {
                    return _.reduce(arr, function (x, y) { return x + y.length; }, 0)
                }
```